hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1386,6 +1387,12 @@ bool MDMParser::registerNet(const char* apn, NetStatus* status, system_tick_t ti
                         break; // force another checkNetStatus() call
                     }
                 }
+                // Query IMSI every 60 sec if not regsitered, to detect IMSI switches
+                if (HAL_Timer_Get_Milli_Seconds() - start_imsi_check > 60000UL) {
+                        start_imsi_check = HAL_Timer_Get_Milli_Seconds();
+                        sendFormated("AT+CIMI\r\n");
+                        waitFinalResp();
```

Comment:
We should probably check for `RESP_OK` here, or if we want to account for `AT+CIMI` potentially failing, at least for `WAIT` (timeout) and goto failure handling code. If second approach, `RESP_OK` in https://github.com/particle-iot/device-os/pull/2179/files#diff-13c08dca4df073df121b6a68266e0105R1446 also needs to be replaced.

Reply:
I made this not check for `RESP_OK` because it's only for debugging and guessed it's probably OK if it doesn't give an OK. While the one here (https://github.com/particle-iot/device-os/pull/2179/files#diff-13c08dca4df073df121b6a68266e0105R1446) is for collecting the IMSI so it has the failure handling code. 
So treating them slightly differently where this one has less importance of getting an answer/OK - may be I am overlooking something. 
If needing consistency, I will have this one also wait for `RESP_OK` and have a failure handing code, but the other block I think is more important and must wait for `RESP_OK`.... 

```
                // Query IMSI every 60 sec if not regsitered, to detect IMSI switches
                if (HAL_Timer_Get_Milli_Seconds() - start_imsi_check > 60000UL) {
                    start_imsi_check = HAL_Timer_Get_Milli_Seconds();
                    sendFormated("AT+CIMI\r\n");
                    if (RESP_OK != waitFinalResp(nullptr, nullptr, CIMI_TIMEOUT)) {
                        goto failure;
                    }
                }
            }
```

Reply:
And will fix formatting :)

Reply:
We still need to confirm that we've gotten at least _some_ response and fail if we didn't (`WAIT`). Otherwise we'll get command requests/responses out of sync.